### DISCUSSION TITLE
Unable to use code-server at all

### DISCUSSION THREAD

**Discussion by ChokunPlayZ** [coder/code-server#2432] [Q&A]
- Web Browser: Firefox
- Local OS: Windows 10 2004
- Remote OS: Ubuntu 18.04.5 LTS (VM)
- Remote Architecture: x86_64
- `code-server --version`: 3.7.4 11f53784c58f68e7f4c5b3b8dae9407caa41725b

I'm able to access the code-server but it says disconnected and it attempting to reconnect but unsuccessful
I know that the problem comes from the server

here is the log I been getting
https://pastebin.com/vPqcYkTw

I tried running both of these commands and still the same problem
`npm rebuild` `npm install`

**Comment by code-asher** [comment]
Could you try going into `lib/vscode` and then running `npm rebuild`?
That should get rid of the version mismatch errors.

This can happen if you run code-server with a different version of Node
than the one it was built with.

**Comment by ChokunPlayZ** [comment]
> 
> Could you try going into `lib/vscode` and then running `npm rebuild`? That should get rid of the version mismatch errors. This can happen if you run code-server with a different version of Node than the one it was built with.

I tried to do rebuild in lib/vscode and this error comes up
[2020-12-10T01_17_06_577Z-debug.log](https://github.com/cdr/code-server/files/5669664/2020-12-10T01_17_06_577Z-debug.log)
https://pastebin.com/5w5DZgkW (its the same file)

and I don't know exactly how to run different version of node
I just installed nodejs with `sudo apt install nodejs` command

**Reply by oxy** [reply]
You need to have at least Node v12 installed - see https://github.com/cdr/code-server/blob/v3.7.4/doc/install.md#yarn-npm.
(Ubuntu 18.04 ships with Node 8)

I'd recommend using the deb we provide, or installing v12 using nodesource or nvm. DigitalOcean has a useful guide: https://www.digitalocean.com/community/tutorials/how-to-install-node-js-on-ubuntu-18-04#installing-using-a-ppa - you can use either the 12.x PPA or NVM.

**Reply by ChokunPlayZ** [reply]
I'm running
npm 6.14.8 and node v14.15.1

should I downgrade node to v12?

**Reply by ChokunPlayZ** [reply]
I just got it working on another fresh VM
I downloaded the new Ubuntu Server 20.04 LTS and installed it on another VM
it's working fine now, not seeing any error
